libsolidity/analysis/PostTypeContractLevelChecker.cpp

```diff
@@ -144,4 +144,47 @@ void PostTypeContractLevelChecker::checkStorageLayoutSpecifier(ContractDefinitio
 			baseSlotExpression.location(),
 			"Contract extends past the end of storage when this base slot value is specified."
 		);
+
+	warnStorageLayoutBaseNearStorageEnd(_contract);
```

Comment:
If you put it here, it's still checked only for contracts with a specifier. It has to be moved out.

We should be seeing this warnings also on contracts with oversized storage but currently they're not there.